policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/md_socsec_subtraction_amount.py

```diff
@@ -12,4 +12,32 @@ class md_socsec_subtraction_amount(Variable):
         "https://www.marylandtaxes.gov/forms/21_forms/Resident_Booklet.pdf#page=14"
     )
     defined_for = StateCode.MD
-    adds = ["taxable_social_security"]
+
+    def formula(person, period, parameters):
+        unit_tss = person.tax_unit("tax_unit_taxable_social_security", period)
+        # allocate unit_tss to head and spouse in proportion to social_security
+        unit_socsec = person.tax_unit("tax_unit_social_security", period)
+        unit_has_socsec = unit_socsec > 0
+        ind_socsec = person("social_security", period)
+        is_spouse = person("is_tax_unit_spouse", period)
+        spouse_frac = min_(
+            1, where(is_spouse & unit_has_socsec, ind_socsec / unit_socsec, 0)
+        )
+        unit_spouse_frac = person.tax_unit.sum(spouse_frac)
+        is_head = person("is_tax_unit_head", period)
+        head_frac = where(is_head & unit_has_socsec, 1 - unit_spouse_frac, 0)
+        unit_head_frac = person.tax_unit.sum(head_frac)
+        """
+        print("\n******************************************")
+        print("unit_tss=", unit_tss)
+        print("unit_socsec", unit_socsec)
+        print("ind_socsec", ind_socsec)
+        print("spouse_frac=", spouse_frac)
+        print("unit_spouse_frac=", unit_spouse_frac)
+        print("head_frac=", head_frac)
+        print("unit_head_frac=", unit_head_frac)
+        """
+        return select(
```

Comment:
See https://github.com/PolicyEngine/policyengine-us/issues/1176
```suggestion
        return unit_tss * select([is_head, is_spouse], [unit_head_frac, unit_spouse_frac], default=0)
```

Reply:
Done.  Thanks for the suggestion.